### ISSUE TITLE
Implementation of k-fold technique

### ISSUE THREAD

**Issue by Saanidhyavats** [intel/dffml#1068]
## Pain Point

K-fold technique method is more robust as compared to other methods for measuring the accuracy of a model. For implementing the idea of Automl, this feature will be beneficial.

## Proposed Solution

What would make things behave most smoothly

**Comment by Saanidhyavats** [comment]
@pdxjohnny I am thinking to work on this. Shall I implement it from scratch?

**Comment by mHash1m** [comment]
Will the conclusion of this issue include something like [`cross_val_score`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_score.html#sklearn-model-selection-cross-val-score)? Maybe returning accuracy for all the folds?
